kernel/main.c

```diff
@@ -200,10 +208,15 @@ read_multiboot_info(void)
       const char *name = TO_PTR(mbi->boot_loader_name);
 
       if (!strcmp(name, "TILCK_EFI")) {
+         struct tilck_extra_boot_info *extra_boot_info = TO_PTR(mbi->apm_table);
 
          printk("Multiboot: detected the TILCK_EFI bootloader\n");
-         printk("Multiboot: ACPI root ptr: %p\n", TO_PTR(mbi->apm_table));
-         acpi_set_root_pointer(mbi->apm_table);
+         printk("Multiboot: ACPI root ptr: %p\n", TO_PTR(extra_boot_info->RSDP));
+         acpi_set_root_pointer(extra_boot_info->RSDP);
+
+         uefi_rt_addr = extra_boot_info->RT;
```

Comment:
Not sure if we're going to use UEFI here in this file or move all the code that calls UEFI to a separate file. So, until we're playing with experiments, that's OK. We'll think about that detail later.